app/models/study_json_record.rb

```diff
@@ -1358,14 +1358,22 @@ def study_references_data
 
     collection = []
     references.each do |reference|
-      collection << {
-                      nct_id: nct_id,
-                      pmid: reference['ReferencePMID'],
-                      reference_type: reference['ReferenceType'],
-                      citation: reference['ReferenceCitation']
-
-                    }
-    end
+      temp = {
+                nct_id: nct_id,
+                pmid: reference['ReferencePMID'],
+                reference_type: reference['ReferenceType'],
+                citation: reference['ReferenceCitation']
+              }
+
+      retractions = reference['RetractionList']
+      temp[:retractions_attributes] = retractions.map do |retraction|
+
+                nct_id: nct_id,
+                pmid: retraction['RetractionPMID'],
+                source: retraction['RetractionSource']
+          end
```

Comment:
We need to check that retractions is not nil before making the call to `map` otherwise we will crash right here

Reply:
revised